data-filter.html

```diff
@@ -741,6 +776,46 @@ <h3 style="text-align: center;">Selected Measurements</h3>
                 this._updateUnits();
             }
 
+            _maxAmountOfMeasurements() {
+              return Math.max(...Object.values(this._attributesGroupedBy).map(it => it.length));
+            }
+
+            _attributesGroupedByObserver(newVal, oldVal) {
+              const max = this._maxAmountOfMeasurements();
+              if (max > 0 && this.selectionUnit.unit === '#') {
+                this.set('selectionUnit.max', max);
+                this.notifyPath('selectionUnit.max');
+              }
+            }
+
+            _computeAttributesGroupedBy(_attributes, _groupedBy, data) {
+              const groupedByAnnotation = {};
+                if (this._groupedBy !== 0) {
+                  const annotationType = this._attributes[this._groupedBy - 1];
+                  data.reduce((acc, attr) => {
+                    if(!acc[this._getSanitizeItemName(attr[annotationType.name])]) {
+                      acc[this._getSanitizeItemName(attr[annotationType.name])] = [];
+                    }
+                    acc[this._getSanitizeItemName(attr[annotationType.name])].push(attr);
+                    return acc;
+                  }, groupedByAnnotation);
+                } else {
+                  groupedByAnnotation['a'] = data;
+                }
+              return groupedByAnnotation;
+            }
+
+            _getAttributes(_attributes) {
+              const filteredAttributes = this._attributes.filter(it => {
+                if (it.name === 'name') {
+                  return false;
+                }
+                if (it.type === 'category')
+                return it.name;
+              })
+              return filteredAttributes.length ? [ {name: "no grouping"}, ...filteredAttributes] : filteredAttributes;
```

Comment:
by default we want to always show `no grouping`, so shouldn't the else part here return only `{name: "no grouping"}` instead of `filteredAttributes`, which will be empty ?